frontend/packages/console-shared/src/components/health-checks/health-checks-alert.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Link } from 'react-router-dom';
+import { Alert, AlertActionCloseButton } from '@patternfly/react-core';
+import { ServiceModel } from '@console/knative-plugin';
+import { STORAGE_PREFIX } from '@console/shared';
+import { K8sResourceKind, K8sKind, referenceForModel } from '@console/internal/module/k8s';
+import {
+  DeploymentConfigModel,
+  DeploymentModel,
+  DaemonSetModel,
+  StatefulSetModel,
+} from '@console/internal/models';
+import './health-checks-alert.scss';
+
+type HealthChecksProps = {
+  resource: K8sResourceKind;
+  model: K8sKind;
+};
+
+const HIDE_HEALTH_CHECK_ALERT_FOR = `${STORAGE_PREFIX}/hide-health-check-alert-for`;
+
+const addHealthChecksRefs = [
+  referenceForModel(DeploymentConfigModel),
+  referenceForModel(DeploymentModel),
+  referenceForModel(DaemonSetModel),
+  referenceForModel(StatefulSetModel),
+  referenceForModel(ServiceModel),
+];
+
+const HealthChecksAlert: React.FC<HealthChecksProps> = ({ resource, model }) => {
+  const [hideHealthCheckAlertFor, setHideHealthCheckAlertFor] = React.useState([]);
+  const containers = resource?.spec?.template?.spec?.containers;
+  const containersName = containers?.map((container) => container.name);
+  const healthCheckAdded = _.every(
+    containers,
+    (container) => container.readinessProbe || container.livenessProbe || container.startupProbe,
+  );
+
+  React.useEffect(() => {
+    const hiddenHealthCheckAlertFor = localStorage.getItem(HIDE_HEALTH_CHECK_ALERT_FOR);
+    if (!_.isEmpty(hiddenHealthCheckAlertFor)) {
+      setHideHealthCheckAlertFor(
+        _.uniq([...hideHealthCheckAlertFor, ...JSON.parse(hiddenHealthCheckAlertFor)]),
+      );
+    }
+  }, [hideHealthCheckAlertFor, resource]);
+
+  const handleAlertAction = (name: string, namespace: string) => {
+    const hideHealthCheckAlert = [...hideHealthCheckAlertFor, `${namespace}/${name}`];
+    setHideHealthCheckAlertFor(hideHealthCheckAlert);
+    localStorage.setItem(HIDE_HEALTH_CHECK_ALERT_FOR, JSON.stringify(hideHealthCheckAlert));
+  };
+
+  return (
+    <>
+      {_.includes(addHealthChecksRefs, referenceForModel(model)) &&
```

Comment:
Probably better to have a utility for this big check. You can have it component scope only.